src/main/java/reposense/parser/ArgsParser.java

```diff
@@ -203,15 +209,54 @@ public static CliArguments parse(String[] args) throws HelpScreenException, Pars
         }
     }
 
+    /**
+     * Returns a {@code Date} that is one month before {@code cliUntilDate} (if present) or one month before report
+     * generation date otherwise.
+     */
+    private static Date getDateMinusAMonth(Optional<Date> cliUntilDate) {
+        Calendar cal = Calendar.getInstance();
+        cliUntilDate.ifPresent(cal::setTime);
+        cal.set(Calendar.HOUR_OF_DAY, 0);
+        cal.set(Calendar.MINUTE, 0);
+        cal.set(Calendar.SECOND, 0);
+        cal.set(Calendar.MILLISECOND, 0);
+        cal.add(Calendar.MONTH, -1);
+        return cal.getTime();
+    }
+
+    /**
+     * Returns date of report generation with time set to 23:59:59.
+     */
+    private static Date getReportGenerationDate() {
+        Calendar cal = Calendar.getInstance();
+        cal.set(Calendar.HOUR_OF_DAY, 23);
+        cal.set(Calendar.MINUTE, 59);
+        cal.set(Calendar.SECOND, 59);
+        cal.set(Calendar.MILLISECOND, 0);
+        return cal.getTime();
+    }
```

Comment:
This is actually, `getCurrentTime` as the method doesn't do anything related to the report.

To consider in future: This kind of utility methods can be moved to a utility class so that they can be reused from elsewhere.